xbmc/application/AppParamParser.cpp

```diff
@@ -57,6 +57,13 @@ void CAppParamParser::Parse(const char* const* argv, int nArgs)
 
   for (int i = 0; i < nArgs; i++)
   {
+#ifdef TARGET_WEBOS
```

Comment:
Platform ifdefs on common code usually is something we try to avoid. 
Maybe create some generic class/file with a preprocessor static method (that does nothing) and is included for all platforms except webos? Then include one that actually does the cleanup for webos?

Reply:
Maybe the platform AppParamParser might be relevant

https://github.com/xbmc/xbmc/blob/master/xbmc/platform/linux/AppParamParserLinux.cpp

Reply:
I'd rather detect the json content and filter it out. That way kodi can actually receive valid args on pos 1. Or is it impossible to call an app with args on webos?

Reply:
Yes, as it's platform specific, this code belongs into https://github.com/xbmc/xbmc/blob/master/xbmc/platform/linux/AppParamParserLinux.cpp, please.

Reply:
Or we even should introduce https://github.com/xbmc/xbmc/blob/master/xbmc/platform/linux/AppParamParserWebOS.cpp as this is not even something general for Linux ?!

Reply:
As a rule of thumb, whenever you introduce an `ifdef platform` in platform agnostic code, you should feel like finding another solution. :-)